dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceFile.java

```diff
@@ -226,6 +234,22 @@ public void setFrozen(boolean frozen) {
         this.frozen = frozen;
     }
 
+    private static void checkPath(String path) {
+        if (path != null && pathRegex != null && !pathRegex.matcher(path).matches()) {
+            throw new CustomWebApplicationException(pathViolationMessage, HttpStatus.SC_BAD_REQUEST);
+        }
+    }
+
+    public static void restrictPaths(Pattern newPathRegex, String newPathViolationMessage) {
+        pathRegex = newPathRegex;
```

Comment:
Yes, technically correct, although the chance of a problem is this particular case is essentially zero.
https://docs.oracle.com/javase/specs/jls/se8/html/jls-17.html#jls-17.4.3
But I added the `volatile`s anyways, why not.
Note: this pattern happens elsewhere in the existing webservice code.

Reply:
Interesting link, I'm not sure though that it offers extra proof as I read it, although it might be too abstract for me to understand correctly:

1. The first part talks about intra-thread, which I think doesn't apply to my example (my example is two threads).
2. The second part talks about sequential consistency, but then goes on to say "If we were to use sequential consistency as our memory model...", which implies they don't use it.

But yes, agreed, the odds are ridiculously slim in the current use case.